usaspending_api/awards/v2/views/funding.py

```diff
@@ -93,7 +93,7 @@
         left outer join submission_attributes sa on
             gfaba.submission_id = sa.submission_id
     group by
```

Comment:
Perhaps I'm not understanding what's supposed to be going on here, but I'm not seeing any aggregations or anything.  Do we need to GROUP?

Reply:
> Not specifically related to this pull request, but what's this?
> 
> ```
> case when sa.subtier_agency_id is null then 1173 else a.id end                   agency_id,
> ```
> 
> Is this intentional?
Yes, according to the design crew, for Army, Navy, and Air Force, we want to link to DoD since we don't have agency pages for those agencies. They're the only agencies that have null subtier codes, so we're just swapping in the DoD id in that case.

Reply:
> Yes, according to the design crew, for Army, Navy, and Air Force, we want to link to DoD since we don't have agency pages for those agencies. They're the only agencies that have null subtier codes, so we're just swapping in the DoD id in that case.

So the number listed there is a surrogate key.  There are no guarantees that it will always be 1137.  We need something a bit more... static.

In the file https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/references/constants.py is the list of toptier cgac codes that map to DoD.  We need to incorporate this list in the query instead of hardcoding the surrogate key.

When I can string two minutes of consecutive thought together, I can craft such a query (unless you want to have a go at it).  We can also simplify the `agency_id_to_agency_id_for_toptier_mapping` bit.   Sadly, that bit of SQL is an artifact of my misunderstanding of how Agencies work.  It works, but is unnecessary.  I have removed those in my branch, but we can preemptively remove it from your SQL to simplify things a bit.